src/lightning/data/streaming/resolver.py

```diff
@@ -0,0 +1,340 @@
+        return _resolve_s3_connections(dir_path_absolute)
+
+    if dir_path_absolute.startswith("/teamspace/datasets") and len(dir_path_absolute.split("/")) > 3:
+        return _resolve_datasets(dir_path_absolute)
+
+    return Dir(path=dir_path_absolute, url=None)
+
+
+def _match_studio(target_id: Optional[str], target_name: Optional[str], cloudspace: V1CloudSpace) -> bool:
+    if cloudspace.name is not None and target_name is not None and cloudspace.name.lower() == target_name.lower():
+        return True
+
+    if target_id is not None and cloudspace.id == target_id:
+        return True
+
+    if (
+        cloudspace.display_name is not None
+        and target_name is not None
+        and cloudspace.display_name.lower() == target_name.lower()
+    ):
+        return True
+
+    return False
+
+
+def _resolve_studio(dir_path: str, target_name: Optional[str], target_id: Optional[str]) -> Dir:
+    client = LightningClient(retry=False)
+
+    # Get the ids from env variables
+    cluster_id = os.getenv("LIGHTNING_CLUSTER_ID", None)
+    project_id = os.getenv("LIGHTNING_CLOUD_PROJECT_ID", None)
+
+    if cluster_id is None:
+        raise RuntimeError("The `cluster_id` couldn't be found from the environement variables.")
+
+    if project_id is None:
+        raise RuntimeError("The `project_id` couldn't be found from the environement variables.")
+
+    clusters = client.cluster_service_list_project_clusters(project_id).clusters
+
+    cloudspaces = client.cloud_space_service_list_cloud_spaces(project_id=project_id, cluster_id=cluster_id).cloudspaces
+    target_cloud_space = [cloudspace for cloudspace in cloudspaces if _match_studio(target_id, target_name, cloudspace)]
+
+    if not target_cloud_space:
+        raise ValueError(f"We didn't find any matching Studio for the provided name `{target_name}`.")
+
+    target_cluster = [cluster for cluster in clusters if cluster.id == target_cloud_space[0].cluster_id]
+
+    if not target_cluster:
+        raise ValueError(
+            f"We didn't find a matching cluster associated with the id {target_cloud_space[0].cluster_id}."
+        )
+
+    bucket_name = target_cluster[0].spec.aws_v1.bucket_name
+
+    return Dir(
+        path=dir_path,
+        url=os.path.join(
+            f"s3://{bucket_name}/projects/{project_id}/cloudspaces/{target_cloud_space[0].id}/code/content",
+            *dir_path.split("/")[4:],
+        ),
+    )
+
+
+def _resolve_s3_connections(dir_path: str) -> Dir:
+    client = LightningClient(retry=False)
+
+    # Get the ids from env variables
+    project_id = os.getenv("LIGHTNING_CLOUD_PROJECT_ID", None)
+    if project_id is None:
+        raise RuntimeError("The `project_id` couldn't be found from the environement variables.")
+
+    target_name = dir_path.split("/")[3]
+
+    data_connections = client.data_connection_service_list_data_connections(project_id).data_connections
```

Comment:
same as above. We could just have those on the project API and they'd basically just be pass-through, but it would be great if we could consolidate the lightning_cloud use

Reply:
Same as above.